### DISCUSSION TITLE
Post Processor is not acting accordingly.

### DISCUSSION THREAD

**Discussion by joy2772** [LuckyPennySoftware/MediatR#965] [Q&A] [answered]
Hello there,

I put a 20sec delay inside the **Post Processor** and the controller is taking 20 sec to send the response!

Is there any to stop the handler from waiting for the **Post Processor**? 
I want to send response to the client before executing the **Post Processor.**

Can anyone help please?

MediatR Version: 12.0.1
Dot Net Core 6 (Web API)

**Startup Class**

`
        services.AddMediatR( cfg => 
            cfg.RegisterServicesFromAssembly(typeof(Query).Assembly));
`

**Handler and PostProcessor**

```

    public class Query : IRequest<Result<Unit>>
    {
        // No params
    }

    public class Handler : IRequestHandler<Query, Result<Unit>>
    {
        public async Task<Result<Unit>> Handle(Query request, CancellationToken cancellationToken)
        {
            return Result<Unit>.Success(Unit.Value);
        }
    }

    public class PostHandler : IRequestPostProcessor<Query, Result<Unit>>
    {
        public async Task Process(Query request, Result<Unit> response, CancellationToken cancellationToken)
        {
            await Task.Delay(20000, cancellationToken);
        }
    }

```

**Comment by jbogard** [comment, marked as answer]
This has nothing to do with MediatR and everything to do with running background tasks. There are simple and complex ways of doing this and it's very dependent on what you want to do. Search for that and you'll find plenty of answers.
